core/src/banking_stage/consumer.rs

```diff
@@ -458,6 +458,84 @@ impl Consumer {
         }
     }
 
+    pub fn process_and_record_aged_transactions(
```

Comment:
this whole fn should be de-dup-ed as much as possible with `process_and_record_transactions`, considering the severity of this fragment of code. Otherwise, it would be pretty easy to update one of the two and forget the other..

Reply:
Rebased after #31542 got merged. This should be addressed now - both paths call into the same internal fn for processing and recording, just with different "pre_results"